### DISCUSSION TITLE
Please add an option to view poll in app.

### DISCUSSION THREAD

**Discussion by Dragonfly-30** [Docile-Alligator/Infinity-For-Reddit#1100] [Ideas]
Currently to view a poll we have to click on "view poll". Please make it like so that we can at least see what options are there. So we don't have to click on the view poll button.

**Comment by patrickdemers6** [comment]
@Wladefant are you okay with it if I begin working on this feature? It is something I have wanted for a while now as well.

**Reply by Wladefant** [reply]
Hey @patrickdemers6 ,
Of course, that would be gladly accepted

**Reply by patrickdemers6** [reply]
After checking out some documentation, it looks like there is no API exposed for interacting with polls. Confirmed by [this](https://www.reddit.com/r/redditdev/comments/sw21o7/using_the_api_to_submit_votes_to_polls/) Reddit post. Unfortunately, until an official API is exposed, we won't be able to add polls to Infinity. :disappointed:

**Reply by Dragonfly-30** [reply]
I don't have much knowledge but how boost does it?

**Reply by Wladefant** [reply]
@Drangonfly30 could you send a screenshot?

**Reply by Wladefant** [reply]
@patrickdemers6 yeah unfortunately

**Comment by Wladefant** [comment]
It seems like the only option would be something like webview as in #1357

**Comment by Dragonfly-30** [comment]
It is just a random screenshot I took from boost![Screenshot_2023-02-05-04-00-10-751.jpeg](https://user-images.githubusercontent.com/113085944/216792202-a604fc5d-c1c4-4fdd-ba96-503671e43ad5.jpeg)

**Reply by Wladefant** [reply]
Interesting, and when you click on view poll?
Can you vote in the app?

**Reply by Dragonfly-30** [reply]
No, it'll take to the website like infinity but it is still convenient.

**Reply by Wladefant** [reply]
Agree

**Reply by us3rname101** [reply]
Quoting myself from an another discussion _it could be done_. Getting poll options is easy enough as they are provided in the json what Infinity when requesting posts. 
Sending a vote is done by graphql, who could've guessed. 
![gJbqwXO](https://user-images.githubusercontent.com/95178906/217010740-15e24ad2-1dbc-4619-8c61-ddfc9905c843.png)
Id is probably as well hardcoded, optionId and postId are both in the json.
